rskj-core/src/main/java/org/ethereum/rpc/JsonRpcWeb3FilterHandler.java

```diff
@@ -0,0 +1,71 @@
+package org.ethereum.rpc;
+
+import co.rsk.rpc.JsonRpcFilterServer;
+import co.rsk.rpc.ModuleDescription;
+import co.rsk.rpc.OriginValidator;
+import com.googlecode.jsonrpc4j.AnnotationsErrorResolver;
+import com.googlecode.jsonrpc4j.DefaultErrorResolver;
+import com.googlecode.jsonrpc4j.MultipleErrorResolver;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.*;
+import org.ethereum.rpc.exception.RskErrorResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URISyntaxException;
+import java.util.List;
+
+/**
+ * Created by ajlopez on 18/10/2017.
+ */
+
+@ChannelHandler.Sharable
+public class JsonRpcWeb3FilterHandler extends SimpleChannelInboundHandler<FullHttpRequest> {
+    private static final Logger LOGGER = LoggerFactory.getLogger("jsonrpc");
+
+    private OriginValidator originValidator;
+
+    public JsonRpcWeb3FilterHandler(String corsOrigins) {
+        try {
+            this.originValidator = new OriginValidator(corsOrigins);
+        } catch (URISyntaxException e) {
+            LOGGER.error("Error creating OriginValidator, origins {}, {}", corsOrigins, e);
+        }
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) throws Exception {
+        HttpMethod httpMethod = request.getMethod();
+        HttpResponse response;
+
+        if (HttpMethod.POST.equals(httpMethod)) {
+            HttpHeaders headers = request.headers();
+
+            String contentType = headers.get(HttpHeaders.Names.CONTENT_TYPE);
+            String origin = headers.get(HttpHeaders.Names.ORIGIN);
+            String referer = headers.get(HttpHeaders.Names.REFERER);
+
+            if (!"application/json".equals(contentType)) {

```

Comment:
I think the default content type for JSON-RPC calls is `application/json-rpc` (ref:  [JsonRpcBasicServer#JSONRPC_CONTENT_TYPE](https://github.com/briandilley/jsonrpc4j/blob/master/src/main/java/com/googlecode/jsonrpc4j/JsonRpcBasicServer.java#L35) )

Reply:
Right but don't break compatibility with existing clients. If this change is applied then support both possibilities (`json` and `json-rpc`)